clang/lib/CIR/CodeGen/CIRGenTypes.cpp

```diff
@@ -168,8 +168,8 @@ mlir::Type CIRGenTypes::convertRecordDeclType(const clang::RecordDecl *RD) {
 
   // Handle forward decl / incomplete types.
   if (!entry) {
-    auto name = getRecordTypeName(RD, "");
-    entry = Builder.getStructTy({}, name, /*body=*/false,
+    auto recordKind = Builder.getRecordKind(RD->getTagKind());
+    entry = Builder.getStructTy(recordKind, {}, RD->getName(), /*body=*/false,
```

Comment:
Perhaps change `getStructTy` API to pass down the `RecordDecl` and call `getRecordKind` inside the implementation?

Reply:
I was doing this initially, but I had some errors regarding "incomplete definition" of the `getRecordKind` method.

I'll give it another shot.

Reply:
I'll leave this as is since `getAnonStruct` wraps around `getStructTy`, and the latter might not have a `RecordDecl`.